src/Workspaces/Core/Portable/Workspace/Solution/Solution.cs

```diff
@@ -2020,11 +2020,28 @@ internal Task<Compilation> GetCompilationAsync(Project project, CancellationToke
                 : SpecializedTasks.Default<Compilation>();
         }
 
-        internal Task<bool> HasCompleteReferencesAsync(Project project, CancellationToken cancellationToken)
+        /// <summary>
+        /// Return reference completeness for the given project.
+        /// </summary>
+        internal Task<bool> IsCompleteAsync(Project project, CancellationToken cancellationToken)
+        {
+            // return IsComplete when compilation is not supported. 
+            // regardless whether project support compilation or not, if projectInfo is not complete, we can't gurantee its reference completeness
+            return project.SupportsCompilation
+                ? this.GetCompilationTracker(project.Id).IsCompleteAsync(this, cancellationToken)
+                : project.IsComplete ? SpecializedTasks.True : SpecializedTasks.False;
```

Comment:
IsComplete is set by Host. so if Host set it true, it will be true otherwise false. in VS host, if intellisesebuild fails, it is set to false. otherwise true.